src/components/asset-list/RecyclerAssetList.js

```diff
@@ -1014,89 +692,50 @@ class RecyclerAssetList extends Component {
       return null;
     }
 
-    const { item = {}, renderItem } = data;
-    const { hideHeader, sections, isCoinListEdited } = this.props;
+    const { sections, isCoinListEdited, nativeCurrency } = this.props;
 
-    if (isCoinListEdited && !(type < 7)) {
+    if (isCoinListEdited && !(type.index < 4)) {
       return null;
     }
 
-    if (type === ViewTypes.HEADER || type === ViewTypes.HEADER_FIRST) {
-      return hideHeader ? null : (
-        <AssetListHeaderRenderer
-          {...data}
-          isCoinListEdited={this.props.isCoinListEdited}
-        />
-      );
-    }
-
-    if (type === ViewTypes.COIN_SAVINGS) {
-      return (
-        <SavingsListWrapper assets={item.assets} totalValue={item.totalValue} />
-      );
-    }
-
-    if (type === ViewTypes.COIN_SMALL_BALANCES) {
-      if (
-        this.renderList.length !== item.assets.length ||
-        smallBalancedChanged
-      ) {
-        smallBalancedChanged = false;
-        const renderList = [];
-        for (let i = 0; i < item.assets.length; i++) {
-          renderList.push(
-            renderItem({
-              item: {
-                ...item.assets[i],
-                isSmall: true,
-              },
-              key: `CoinSmallBalances${item.assets[i].symbol}`,
-            })
-          );
-        }
-        this.renderList = renderList;
-      }
-
-      return <SmallBalancesWrapper assets={this.renderList} />;
-    }
-
-    if (type === ViewTypes.COIN_DIVIDER) {
-      return (
-        <CoinDivider
-          assetsAmount={item.assetsAmount}
-          balancesSum={item.value}
-          isCoinListEdited={isCoinListEdited}
-          nativeCurrency={this.props.nativeCurrency}
-        />
-      );
+    if (type.index === ViewTypes.HEADER.index) {
```

Comment:
do we not need to include logic for if the index is ViewTypes.FOOTER ?

Reply:
Not really - If the component doesn't have renderer rowRenderer doesn't have to return anything - for footer we wanted only to take some space - LayoutProvider above made that happen

We can just make rowRenderer return null for cleanliness